sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/ReadSchemaTest.scala

```diff
@@ -72,6 +72,58 @@ trait ReadSchemaTest extends QueryTest with SharedSparkSession {
   val options: Map[String, String] = Map.empty[String, String]
 }
 
+trait ArrayWithNestedStructTest extends ReadSchemaTest {
```

Comment:
The goal of `ReadSchemaTest` is the following (https://github.com/apache/spark/pull/29573/files#diff-ff46c4f16bf7cb1f09e74b70b0ba3bc6be62203824a2fe16c4c473e34fd18d7fR24).
```
Read schema suites have the following hierarchy and aims to guarantee users a backward-compatible read-schema change coverage
```
To simply put, it was about a kind of acceptable schema evolution. However, the new test suites `ArrayWithNestedStructTest` and `MapWithNestedStructTest` seems to have no schema evolution.